metabase/metabase
# Show custom warning message when leaving editing model via Cancel button

Closes #34366

### Description

- Similar to #34405

### How to verify

Prerequisite for all scenarios:
1. Create and save a model

**1. Custom warning modal is shown when leaving editing model via Cancel button with unsaved changes**
1. Click "Edit query definition" or "Edit metadata"
2. Change something (the query, column metadata, etc.)
3. Click "Cancel"

Expected result: custom warning modal is **shown**.

**2. Custom warning modal is not shown when leaving editing model via Cancel button with no changes**
1. Click "Edit query definition" or "Edit metadata"
2. Click "Cancel"

Expected result: custom warning modal is **not shown**.

### Demo


https://github.com/metabase/metabase/assets/6830683/21eba361-e472-4186-af18-85442246aae1
